client/components/header-cake/index.jsx

```diff
@@ -1,35 +1,34 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-var Card = require( 'components/card' ),
-	Gridicon = require( 'components/gridicon' );
+import Card from 'components/card';
+import HeaderCakeBack from './back';
+
+export default React.createClass( {
 
-module.exports = React.createClass( {
 	displayName: 'HeaderCake',
 
 	propTypes: {
-		onClick: React.PropTypes.func.isRequired,
-		onTitleClick: React.PropTypes.func,
-		backText: React.PropTypes.oneOfType( [
-			React.PropTypes.element,
-			React.PropTypes.string
-		] )
+		onClick: PropTypes.func.isRequired,
+		onTitleClick: PropTypes.func,
+		backText: PropTypes.string
 	},
 
-	getDefaultProps: function() {
+	getDefaultProps() {
 		return {
 			isCompact: false
 		};
 	},
 
-	render: function() {
-		var classes = classNames(
+	render() {
+		const backText = this.props.backText || this.translate( 'Back' );
```

Comment:
Might want to take advantage of the soft-failing destructuring assignment here, maybe.

```js
const { backText = this.translate( 'Back' ) } = this.props;
```